python/ray/train/lightning/_lightning_utils.py

```diff
@@ -264,10 +277,15 @@ def on_train_epoch_end(self, trainer, pl_module) -> None:
 
         # Save checkpoint to local
         ckpt_path = os.path.join(tmpdir, "checkpoint.ckpt")
+
+        # Lightning determines saving on rank zero or all ranks
         trainer.save_checkpoint(ckpt_path, weights_only=False)
 
         # Report to train session
-        checkpoint = Checkpoint.from_directory(tmpdir)
+        if self.rank_zero_only and self.world_rank != 0:
```

Comment:
Should all the checkpoint creation logic above be within this if?

Reply:
Actually we don't need to, because lightning would decide whether to save which rank(s). It saves only on rank 0 with DDP, on all ranks with zero-3